clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResult.java

```diff
@@ -56,4 +68,14 @@ public boolean doesResultReuseInput() {
   public GenericRecord getRmdRecord() {
     return rmdRecord;
   }
+
+  /**
+   * Provide the deserialized new value on a best-effort approach. Meaning that it's acceptable to return an empty
+   * Optional. e.g. MergeConflictResult of full PUTs will not contain deserialized new value since we don't need to
+   * deserialize the value to generate the MCR.
+   * @return deserialized new value if possible.
+   */
+  public Optional<GenericRecord> getNewValueDeserialized() {
```

Comment:
What is the expected type of this GenericRecord?  I'm assuming it's the user record sans Venice enveloping?

Reply:
It should be the user value record without the Venice enveloping, see integration test example.